nomt/src/beatree/ops/overflow.rs

```diff
@@ -93,18 +93,25 @@ pub fn decode_cell<'a>(raw: &'a [u8]) -> (usize, [u8; 32], impl Iterator<Item =
     assert!(raw.len() >= 8 + 4 + 32);
     assert_eq!(raw.len() % 4, 0);
 
-    let value_size = u64::from_le_bytes(raw[0..8].try_into().unwrap());
+    let value_size = u64::from_le_bytes(raw[0..8].try_into().unwrap()) as usize;
+    // SAFETY: values bigger than MAX_OVERFLOW_VALUE_SIZE are not allowed.
+    assert!(value_size < MAX_OVERFLOW_VALUE_SIZE);
```

Comment:
To align with the comment and the logic in `encode_cell` this should be:

```suggestion
    assert!(value_size <= MAX_OVERFLOW_VALUE_SIZE);
```